core/store/migrate/migrate.go

```diff
@@ -103,7 +103,9 @@ func ensureMigrated(db *sql.DB, lggr logger.Logger) {
 
 func Migrate(db *sql.DB, lggr logger.Logger) error {
 	ensureMigrated(db, lggr)
-	return goose.Up(db, MIGRATIONS_DIR)
+	// WithAllowMissing is necessary when upgrading from 0.10.14 since it
+	// includes out-of-order migrations
+	return goose.Up(db, MIGRATIONS_DIR, goose.WithAllowMissing())
```

Comment:
Curious, based on a previous [comment](https://github.com/smartcontractkit/chainlink/pull/5578#issuecomment-983822064) this was not required.

Why did you end up having to add this option?